### PR TITLE
feat(useVirtualList): horizontal list

### LINKED ISSUE #2285: horizontal `useVirtualList`
### Clear and concise description of the problem

`useVirtualList` is great! It would be great if we could use this for horizontal scroll as well.

### Suggested solution

Add `horizontal` support

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by Surof1n**
### Description

add horizontal scroll to useVirtualList, fixes #2285 

### Additional context

I would like to know if I was moving in the right direction for a solution. Thanks

### What is the purpose of this pull request?  

- [ ] Bug fix
- [x] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Comment by Surof1n** [comment]
Single question, maybe we don't should setting display: flex for container

**Comment by Surof1n** [comment]
[antfu](https://github.com/antfu) Hello, can you tell me if this PR can be expected to be verified, or is something else needed?

**Comment by Surof1n** [comment]
@antfu Hello, sorry for the link, can you tell me if this PR can be expected to be verified, or is something else needed?

File: packages/core/useVirtualList/index.ts [hunk 1]
```diff
@@ -147,18 +309,12 @@ export function useVirtualList <T = any>(list: MaybeRef<T[]>, options: UseVirtua
...
-      ref: containerRef,
-      onScroll: () => {
-        calculateRange()
-      },
-      style: containerStyle,
-    },
+    containerStyle,
     wrapperProps,
+    currentList,
+    containerRef,
```
Comment (antfu) [inline packages/core/useVirtualList/index.ts:318]: The return types seems introduce a breaking change, is it intentional?

Comment (Surof1n) [reply packages/core/useVirtualList/index.ts:318]: If you look at this feature, you will notice that there are no breaking changes.
```js
export function useVirtualList <T = any>(list: MaybeRef<T[]>, options: UseVirtualListOptions): UseVirtualListReturn<T> {
  const { containerStyle, wrapperProps, scrollTo, calculateRange, currentList, containerRef } = 'itemHeight' in options
    ? useVerticalVirtualList(options, list)
    : useHorizontalVirtualList(options, list)

  return {
    list: currentList,
    scrollTo,
    containerProps: {
      ref: containerRef,
      onScroll: () => {
        calculateRange()
      },
      style: containerStyle,
    },
    wrapperProps,
  }
}
```

The original function still returns the same object as before.

It's just that the object has different values ​​depending on the direction of the list passed, but nothing has changed for the vertical sheet.

**Pull request merged**